vortex-serde/src/layouts/read/mod.rs

```diff
@@ -53,56 +56,27 @@ pub enum ReadResult {
     Batch(Array),
 }
 
+#[derive(Debug)]
+pub enum RangeResult {
+    ReadMore(Vec<Message>),
+    Rows(Option<RowSelector>),
+}
+
 pub trait LayoutReader: Debug + Send {
+    /// Produce sets of row ranges to read from underlying layouts.
```

Comment:
This doc talks about sets and empty, but the actual RangeResult is an Option now.